sys/shell/cmds/cord_ep.c

```diff
@@ -51,23 +52,48 @@ static int _cord_ep_handler(int argc, char **argv)
     int res;
 
     if ((argc > 1) && (strcmp(argv[1], "register") == 0)) {
-        char *regif = NULL;
         if (argc < 3) {
-            printf("usage: %s register <server address> [registration interface]\n",
+            printf("usage: %s register <registration uri>\n",
                    argv[0]);
+            printf("example: %s register coap://[2001:DB8::1]:99/login\n", argv[0]);
             return 1;
         }
+
+        uri_parser_result_t uri_result;
+        if (uri_parser_process_string(&uri_result, argv[2]) != 0) {
+            puts("error: unable to parse uri");
+            return 1;
+        }
+
+        if ((uri_result.scheme_len != 4) || (strncmp(uri_result.scheme, "coap", 4) != 0)) {
+            puts("error: Only coap schemes are supported");
+            return 1;
+        }
+
+        if (uri_result.ipv6addr == NULL) {
+            puts("error: Only ipv6 addresses are supported");
+            return 1;
+        }
+
         sock_udp_ep_t remote;
-        if (make_sock_ep(&remote, argv[2]) < 0) {
-            printf("error: unable to parse address\n");
+        remote.family = AF_INET6;
+        remote.netif = SOCK_ADDR_ANY_NETIF;
+        if (!ipv6_addr_from_buf((ipv6_addr_t *)&remote.addr.ipv6, uri_result.ipv6addr,
+                                uri_result.ipv6addr_len)) {
+            puts("error: IPv6 address malformed");
             return 1;
         }
-        if (argc > 3) {
-            regif = argv[3];
+
+        if (uri_result.port == 0) {
+            remote.port = COAP_PORT;
         }
+        else {
+            remote.port = uri_result.port;
+        }
+
         puts("Registering with RD now, this may take a short while...");
-        if (cord_ep_register(&remote, regif) != CORD_EP_OK) {
-            puts("error: registration failed");
+        if (cord_ep_register(&remote, uri_result.path) != CORD_EP_OK) {
```

Comment:
This has the subtle behavior quirk that when an empyt-path URI is passed in without a trailing slash (eg. `coap://[2001:db8::1]`), autio-discovery is performed, whereas when passed in with a trailing slash (`coap://[2001:db8::1]/`), it is taken literally as the registration resource.

I'd consider this a feature even, if not RFC7252 declared that the URIs equivalent in this protocol. I wouldn't do anything about that here -- maybe it's a feature even despite the equivalence.